### DISCUSSION TITLE
is there a way to pre load a table into RAM

### DISCUSSION THREAD

**Discussion by djouallah** [duckdb/duckdb#4423] [Q&A]
using python, is there a way to load a table and keep it in memory, I know duckdb is in-process so when the python process close it will automatically close DuckDB, any hack maybe ?

**Comment by Alex-Monahan** [comment]
All temporary tables are stored in RAM (unless they are larger than RAM, then k think they spill to disk). So you could do:
```sql
create temp table ram_cache as
select * from data_i_want_in_ram
```
Once the database connection is closed, temp tables disappear.

**Reply by djouallah** [reply]
@Alex-Monahan I want somehow to keep it in memory even when the connection is closed ?
